lib/SILOptimizer/Transforms/GenericSpecializer.cpp

```diff
@@ -129,6 +220,57 @@ bool GenericSpecializer::specializeAppliesInFunction(SILFunction &F) {
   return Changed;
 }
 
+/// Check if any argument to a function meet the criteria for devirtualization.
+bool ProtocolDevirtualizer::canDevirtualizeProtocolInFunction(ProtocolDevirtualizerAnalysis *PDA,
+        llvm::SmallDenseMap<int, std::pair<ProtocolDecl*, ClassDecl *>> &Arg2DeclMap) {
+  auto *F = getFunction();
+  auto Args = F->begin()->getFunctionArguments();
+  bool returnFlag = false;
+
+  /// Analyze the argument for protocol conformance.
+  for (unsigned i = 0, e = Args.size(); i != e; ++i) {
+    auto ArgType = Args[i]->getType();
+    /// Keep it simple for now. Do not handle Protocol constrained methods or Optionals.
+    auto SwiftArgType = ArgType.getSwiftRValueType();
+    if (ArgType && ArgType.isAnyExistentialType()) {
```

Comment:
isAnyExistentialType() is also true for existential metatypes. If you don't care about those, you can use isExistentialType() instead.